### ISSUE TITLE
SA1106 should be reported for the semicolon following a type declaration

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#1254]
The following is legal C# code:

```csharp
class TypeName
{
};
```

However, the `;` at the end is not necessary. It should be treated as an empty statement for the purpose of SA1106.

**Comment by pdelvo** [comment]
grabbed
